### DISCUSSION TITLE
New board and connectors on replacement S100105

### DISCUSSION THREAD

**Discussion by kenmaples** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#810] [General]
I built a module for my pump that was being replaced (due to recall replacement), so I had the unit here to play with. Everything seemed to be going well until I opened the new unit to do the final install.
The new S100105 unit has a new board and connector. The console\display now connects directly to the board.

![IMG_7382](https://github.com/user-attachments/assets/07084fbb-5036-4886-a648-2822d61461e8)

![IMG_7380](https://github.com/user-attachments/assets/1c55e304-2bc4-404f-a5d8-3632df3b5c5b)

**Comment by visualapproach** [comment]
That's a bummer

**Comment by kenmaples** [comment]
I hastily threw in some jumpers on both ends of the module I made, and everything is up and running as expected so far.

**Reply by mossc001** [reply]
The new connectors are JST-XH if you don't want to make the lead; https://www.amazon.co.uk/dp/B0CBWY23FN

**Comment by BigWebstas** [comment]
@kenmaples I have the same new board with the new connector, I am able to get the esp to run the pump/blower etc but i cant get the display to function can i ask what your setup is?

**Reply by ericreich** [reply]
I am in similar situation with new connectors.  I used the new connector, but not getting anything on the display.

**Comment by BigWebstas** [comment]
I have tried everything it seems but the display just does this
![image](https://github.com/user-attachments/assets/2c8fbfe5-1285-4d44-829b-47a7d110d26e)
Heres some shots of the new plug, it plugs straight into the pump board, they also used grey silicone so you cant see the board components or board revision :(
![image](https://github.com/user-attachments/assets/2c47b8fb-b7d9-4192-b240-779984f31ca5)
![image](https://github.com/user-attachments/assets/d15b24ba-3f66-46b2-a4ee-5eca0933fbf7)
![image](https://github.com/user-attachments/assets/4c5cbc6d-98e2-4b1a-8016-0e055e98cdea)
![image](https://github.com/user-attachments/assets/f602d6f7-d0fa-4356-99f1-505a0a7099c7)

**Comment by kenmaples** [comment]
@BigWebstas I didn't have to do anything additional with the new board other than throw in the updated connectors. Are you working with a S100105? The board, ignoring all that fantastic epoxy, looks a bit different, Mine is a 6 wire install that I used a D1 mini for.

**Reply by BigWebstas** [reply]
Mine has 4 wires from the display but the pump has 6 pins.   Its a S100205 im guessing the 2 means it has built in wifi, I am thinking that is the issue. I can get it to control the pump but the display does nothing.  Thanks for the info

**Comment by rmchik** [comment]
@kenmaples , I assembled 2 units on D1mini for me and neighbour and my friend recently purchased pool with the board just like yours, 6-wire

I made adapters for new connectors, keeping pin order (at1st step finding 5v and ground), and.. display doesn't turns on :-(
D1 mini gets power, turns on, connects to wifi, passes hw test  with adapters attached, but it seems that pionout on new JST-XH connector is not the same. 6 wires, 2 known as +5 and GND.. that's 24 combintaions to try :)

**Reply by visualapproach** [reply]
Hi, if you do an input test you should be able to get an idea of the pinout. Go to layzspa.local/inputs/ (IIRC) or check /links.html
CS has fewest changes, CLK has most

**Reply by hmpff1990** [reply]
Hi all,
So if i replace those connectors to xh they remain the same?
Ive got the v2b board and il just replace it with xh pin header and 1 cable to connect to the logic board?

**Reply by rmchik** [reply]
> Hi, if you do an input test you should be able to get an idea of the pinout. Go to layzspa.local/inputs/ (IIRC) or check /links.html CS has fewest changes, CLK has most

Hi! Maybe, you can help me with it? Not sure I can handle it myself..

[inputs.zip](https://github.com/user-attachments/files/20444337/inputs.zip)

Now it looks like this:

E37035  -    2 entries
E35035  -   17 entries
E33035  -  127 entries
E32035  -   91 entries
E31035  -  412 entries
E30035   -  375 entries

**Reply by hmpff1990** [reply]
I took over 1 on 1 jst sm to xh and it works fine,
Got it connected to mqtt and on homebridge, like easy peasy 

model 100101 newest logic board

![IMG_6166](https://github.com/user-attachments/assets/5fcdfba7-2974-4e62-8482-fa71f8804d2e)
![IMG_6154](https://github.com/user-attachments/assets/269964e1-8273-47ac-8973-fc72c68a8d8c)

**Reply by hmpff1990** [reply]
[inputs.zip](https://github.com/user-attachments/files/20444776/inputs.zip)

here are the inputs from the 100101

**Reply by rmchik** [reply]
> [inputs.zip](https://github.com/user-attachments/files/20444776/inputs.zip)
> 
> here are the inputs from the 100101

file inside archive has a 0 byte size

**Reply by rmchik** [reply]
> [inputs.zip](https://github.com/user-attachments/files/20444776/inputs.zip)
> 
> here are the inputs from the 100101

btw, my box has 100101 p/n too

maybe, I should try another esp board, i have a spare.. there was a case when it solved similiar problem

**Reply by rmchik** [reply]
> Hi, if you do an input test you should be able to get an idea of the pinout. Go to layzspa.local/inputs/ (IIRC) or check /links.html CS has fewest changes, CLK has most

During a fit of despair I've tried to change hw config from cio pins 4,3,2 and dsp 7,6,5,0 to 7,6,5 and 4,3,2,0.

I've got display showing scrolling text on ESP startup, a temperature showing 27, I'm able to change brightnes... ands that's all, nothing else is working.

I've double checked wiring and soldering..

**Reply by hmpff1990** [reply]
Picture of the board?

**Reply by rmchik** [reply]
> Picture of the board?

when hw config set to: 4,3,2/7,6,5,0, display is dead
when set to 7,6,5/4,3,2,0, display show temperature and i cat set brightness via interface, it scrolls firmware version on startup, and it's all

![photo_2025-05-27_13-42-25 (2)](https://github.com/user-attachments/assets/db6c4178-74fd-4bff-b6b3-4ef7e0022baa)
![photo_2025-05-27_13-42-25](https://github.com/user-attachments/assets/89ca667a-2fb3-4f6d-b8f0-5d85f3d963be)

**Reply by hmpff1990** [reply]
I think u can better get the pcb and the whole thing easier to work with got 1 in spare
